### ISSUE TITLE
[x86/Windows] Failure in JIT\HardwareIntrinsics\X86\Sse2\MaskMove_r\MaskMove_r.cmd

### ISSUE THREAD

**Issue by BruceForstall** [dotnet/coreclr#19000]
x86 Windows JitStress=2 JitStressRegs=3 failure

https://ci.dot.net/job/dotnet_coreclr/job/master/view/x86/job/jitstress/job/x86_checked_windows_nt_jitstress2_jitstressregs3/555/consoleText

```
  BEGIN EXECUTION
         "D:\j\workspace\x86_checked_w---508bfa85\tests\..\bin\tests\Windows_NT.x86.Checked\Tests\Core_Root\corerun.exe" MaskMove_r.exe 
        Expected: 100
        Actual: 85026320
        END EXECUTION - FAILED
        FAILED
        Test Harness Exitcode is : 1
        
  To run the test:
  > set CORE_ROOT=D:\j\workspace\x86_checked_w---508bfa85\bin\tests\Windows_NT.x86.Checked\Tests\Core_Root
  > D:\j\workspace\x86_checked_w---508bfa85\bin\tests\Windows_NT.x86.Checked\JIT\HardwareIntrinsics\X86\Sse2\MaskMove_r\MaskMove_r.cmd
  
        Expected: True
        Actual:   False
          D:\j\workspace\x86_checked_w---508bfa85\bin\tests\Windows_NT.x86.Checked\TestWrappers\JIT.HardwareIntrinsics\JIT.HardwareIntrinsics.XUnitWrapper.cs(14737,0): at JIT_HardwareIntrinsics._X86_Sse2_MaskMove_r_MaskMove_r_._X86_Sse2_MaskMove_r_MaskMove_r_cmd()
```

Same test failure previously: https://github.com/dotnet/coreclr/issues/17634

**Comment by BruceForstall** [comment]
@tannergooding @fiigii @CarolEidt 
cc @dotnet/jit-contrib

**Comment by CarolEidt** [comment]
@4creators - since this involves `COMPlus_JitStressRegs=3`, it may be a problem in either `Lowering` or `LSRA`. Ping me if you have questions.

**Comment by 4creators** [comment]
@CarolEidt Thnx for the tip

**Comment by 4creators** [comment]
x86 Windows JitStress=2 JitStressRegs=3 MoveMask_r test failure is a result of a regression introduced by PR #16517 

Via `git bisect`:

commit b39a5b2cfad4620ae23cda84186a36ffb463cd5a fails
commit 74d0196fe5b36bf04848f2cf84d20c8b6e999b62 passes

I have attached disasms and dumps for both b39a5b2cfad4620ae23cda84186a36ffb463cd5a and 74d0196fe5b36bf04848f2cf84d20c8b6e999b62 commits - failing ones have `Error` in the file name

[MaskMove_r_Disasm_Error.txt](https://github.com/dotnet/coreclr/files/2216938/MaskMove_r_Disasm_Error.txt)
[MaskMove_r_Disasm_NoDiffable_Error.txt](https://github.com/dotnet/coreclr/files/2216939/MaskMove_r_Disasm_NoDiffable_Error.txt)
[MaskMove_r_Dump_Error.txt](https://github.com/dotnet/coreclr/files/2216942/MaskMove_r_Dump_Error.txt)

[MaskMove_r_Disasm.txt](https://github.com/dotnet/coreclr/files/2216986/MaskMove_r_Disasm.txt)
[MaskMove_r_Disasm_NoDiffable.txt](https://github.com/dotnet/coreclr/files/2216987/MaskMove_r_Disasm_NoDiffable.txt)
[MaskMove_r_Dump.txt](https://github.com/dotnet/coreclr/files/2216988/MaskMove_r_Dump.txt)

The diffs between correct code gen and one containing error(s) seem to be unrelated to HW Intrinsics but rather arise in intrinsic surrounding code region and at the end of the function.

```asm
       // Original assembly around MaskMove intrinsic
       lea      ecx, bword ptr [ebp-64H]
       call     [TestTable`1[SByte][System.SByte]:get_outArrayPtr():int:this]
       vmovupd  xmm0, xmmword ptr [ebp-94H]
       vmovupd  xmm1, xmmword ptr [ebp-74H]
       mov      edi, eax
       vmaskmovdqu xmm0, xmm1
       mov      eax, gword ptr [classVar[0xd1ffab1e]]
       lea      edi, bword ptr [ebp-64H]
       mov      ecx, eax
       or       ecx, eax
       jne      SHORT G_M47968_IG193
       mov      ecx, 0xD1FFAB1E
       call     CORINFO_HELP_NEWSFAST
       mov      ecx, gword ptr [classVar[0xd1ffab1e]]
       push     0xD1FFAB1E
       mov      edx, ecx
       mov      gword ptr [ebp-B0H], eax
       mov      ecx, eax
       call     System.MulticastDelegate:CtorClosed(ref,int):this
       lea      edx, [classVar[0xd1ffab1e]]
       mov      eax, gword ptr [ebp-B0H]
       call     CORINFO_HELP_CHECKED_ASSIGN_REF_EAX
       mov      ecx, edi
       mov      edx, eax
       call     [TestTable`1[SByte][System.SByte]:CheckResult(ref):bool:this]
       test     eax, eax
```
```asm
       // Disasm with changes introduced by PR #16517 
       lea      ecx, bword ptr [ebp-64H]
       call     [TestTable`1[SByte][System.SByte]:get_outArrayPtr():int:this]
       vmovupd  xmm0, xmmword ptr [ebp-94H]
       vmovupd  xmm1, xmmword ptr [ebp-74H]
       mov      edi, eax
       vmaskmovdqu xmm0, xmm1
       mov      eax, gword ptr [classVar[0xd1ffab1e]]
       lea      ecx, bword ptr [ebp-64H]
       mov      bword ptr [ebp-B0H], ecx
       mov      ecx, eax
       or       ecx, eax
       jne      SHORT G_M47968_IG195
       mov      ecx, 0xD1FFAB1E
       call     CORINFO_HELP_NEWSFAST
       mov      ecx, gword ptr [classVar[0xd1ffab1e]]
       push     0xD1FFAB1E
       mov      edx, ecx
       mov      gword ptr [ebp-B4H], eax
       mov      ecx, eax
       call     System.MulticastDelegate:CtorClosed(ref,int):this
       lea      edx, [classVar[0xd1ffab1e]]
       mov      eax, gword ptr [ebp-B4H]
       call     CORINFO_HELP_CHECKED_ASSIGN_REF_EAX
       mov      ecx, bword ptr [ebp-B0H]
       mov      edx, eax
       call     [TestTable`1[SByte][System.SByte]:CheckResult(ref):bool:this]
       test     eax, eax
```

```asm
      // Original assembly at the end of MaskMove test function
       call     [System.Console:WriteLine()]
       xor      eax, eax
       xor      edi, edi
       mov      dword ptr [ebp-20H], edi
       lea      ecx, bword ptr [ebp-64H]
       call     [TestTable`1[SByte][System.SByte]:Dispose():this]
       jmp      SHORT G_M47968_IG259
       lea      ecx, bword ptr [ebp-64H]
       call     [TestTable`1[SByte][System.SByte]:Dispose():this]
       pop      eax
       jmp      eax
       mov      eax, dword ptr [ebp-20H]
       lea      esp, [ebp-08H]
       pop      esi
       pop      edi
       pop      ebp
       ret      
```

```asm
       // Disasm with changes introduced by PR #16517
       call     [System.Console:WriteLine()]
       xor      edi, edi
       lea      ecx, bword ptr [ebp-64H]
       call     [TestTable`1[SByte][System.SByte]:Dispose():this]
       jmp      SHORT G_M47968_IG259
       lea      ecx, bword ptr [ebp-64H]
       call     [TestTable`1[SByte][System.SByte]:Dispose():this]
       pop      eax
       jmp      eax
       mov      eax, edi
       lea      esp, [ebp-08H]
       pop      esi
       pop      edi
       pop      ebp
       ret 
```

**Comment by 4creators** [comment]
cc @CarolEidt @BruceForstall  @mikedn @tannergooding

**Comment by 4creators** [comment]
@CarolEidt @BruceForstall Problem was fixed some time ago. We can close this issue.

**Comment by tannergooding** [comment]
@4creators, do you have a link to the commit that resolved the issue?

**Comment by 4creators** [comment]
@CarolEidt fixed this issue in #18941, however, this was masked by other problems which were introduced with this PR.
